src/models/CommonModel.ts

```diff
@@ -41,7 +137,7 @@ export class CommonModel extends CommonSchema<CommonModel> {
    * @param mergeFrom 
    * @param originalSchema 
    */
-  private static mergeProperties(mergeTo: CommonModel, mergeFrom: CommonModel, originalSchema: Schema) {
+  private static mergeProperties(mergeTo: CommonModel, mergeFrom: CommonModel, originalSchema: Schema, alreadyIteratedModels: Map<CommonModel, CommonModel> = new Map()) {
```

Comment:
There is so many arguments here that I would consider passing them in a object. If you would do it this way, you dont have to care about order while developing.

Reply:
I tried and I agree, also I think it is time for the merge functions to get their own file. However, that is too much of a refactor in this PR (trying to keep it as small as possible). Would you mind adding an issue for this or should I?

Reply:
https://github.com/asyncapi/generator-model-sdk/issues/191